adapters/consumable/consumable.go

```diff
@@ -0,0 +1,288 @@
+		Time:               a.clock.Now().Unix(),
+		IncludePricingData: true,
+		EnableBotFiltering: true,
+		Parallel:           true,
+	}
+
+	if request.Site != nil {
+		body.Referrer = request.Site.Ref // Effectively the previous page to the page where the ad will be shown
+		body.Url = request.Site.Page     // where the impression will be made
+	}
+
+	for i, impression := range request.Imp {
+
+		_, consumableExt, err := extractExtensions(impression)
+
+		if err != nil {
+			return nil, err
+		}
+
+		// These get set on the first one in observed working requests
+		if i == 0 {
+			body.NetworkId = consumableExt.NetworkId
+			body.SiteId = consumableExt.SiteId
+			body.UnitId = consumableExt.UnitId
+			body.UnitName = consumableExt.UnitName
+		}
+
+		body.Placements[i] = placement{
+			DivName:   impression.ID,
+			NetworkId: consumableExt.NetworkId,
+			SiteId:    consumableExt.SiteId,
+			UnitId:    consumableExt.UnitId,
+			UnitName:  consumableExt.UnitName,
+			AdTypes:   getSizeCodes(impression.Banner.Format), // was adTypes: bid.adTypes || getSize(bid.sizes) in prebid.js
+		}
+	}
+
+	bodyBytes, err := json.Marshal(body)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	fmt.Printf("%s", bodyBytes)
+	requests := []*adapters.RequestData{
+		{
+			Method:  "POST",
+			Uri:     "https://e.serverbid.com/api/v2",
+			Body:    bodyBytes,
+			Headers: headers,
+		},
+	}
+
+	return requests, nil
+}
+
+/*
+internal original request in OpenRTB, external = result of us having converted it (what comes out of MakeRequests)
+*/
+func (a *ConsumableAdapter) MakeBids(
+	internalRequest *openrtb.BidRequest,
+	externalRequest *adapters.RequestData,
+	response *adapters.ResponseData,
+) (*adapters.BidderResponse, []error) {
+
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+
+	var serverResponse bidResponse // response from Consumable
+	if err := json.Unmarshal(response.Body, &serverResponse); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("error while decoding response, err: %s", err),
+		}}
+	}
+
+	bidderResponse := adapters.NewBidderResponse()
+	var errors []error
+
+	for impID, decision := range serverResponse.Decisions {
+		imp := getImp(impID, internalRequest.Imp)
+		if imp == nil {
+			errors = append(errors, &errortypes.BadServerResponse{
+				Message: fmt.Sprintf("ignoring bid id=%s, request doesn't contain any impression with id=%s", "TODO: bid.ID", impID),
+			})
+			fmt.Printf("%s", errors[0])
```

Comment:
We should drop this error print ... the error should be visible in the PBS response and it is not clear that this is actionable by the PBS host.